### DISCUSSION TITLE
[Form] [Feature proposal] Add isDisabled to enum options

### DISCUSSION THREAD

**Discussion by mathieutu** [filamentphp/filament#15048] [Ideas]
Hi, 
Like `getDescription` and `getLabel`, I'd like to add a `CanBeDisabled` contract with a `isDisabled` method to enums that are used as options.

I'm ready to make the PR, would you accept it?

I thought about using `configureUsing`, but as we can't directly access the `options` property, we can't check if it's an enum, and get which enum.

cc @danharrin

Thanks!

_PS: opening this discussion first, like required [here](https://filamentphp.com/docs/3.x/support/contributing#development-of-new-features)_

**Comment by danharrin** [comment]
Hmm, that does feel a little specific per-select. What use case do you have for it?

**Reply by mathieutu** [reply]
For now, I'm just presenting options that are not implemented yet, so not available (depending on feature flags).

But it allows deporting the logic of what can be selected outside the Form configuration. The Enum owns the responsibility to know which options are available or not. (and it can implement any logic for that)

**Reply by mathieutu** [reply]
BTW, I'm wondering about also having a static `getDefault`.

This could be implemented with `configureUsing` if we had a `$comp->getEnum` method or something equivalent.

**Reply by danharrin** [reply]
Yeah sorry but I am still not convinced, usually when we implement enum interfaces they can be used across more than just 1 component

**Reply by mathieutu** [reply]
Ok, thanks for the answer!

**Comment by mathieutu** [comment]
Will not be implemented.

For the record, I've pushed it even further for my case and added a macro:

```php
Radio::macro('fromEnum', function (string $enumClass) {
    if (! is_a($enumClass, FilamentEnum::class, true)) {
        throw new InvalidArgumentException("The class $enumClass must implements FilamentEnum");
    }

    return self::make($enumClass::getKey())
        ->options($enumClass)
        ->default($enumClass::getDefault()?->value)
        ->disableOptionWhen(fn (string $value) => (bool) $enumClass::from($value)->isDisabled());
};
```

With 
```php
interface FilamentEnum extends HasLabel
{
    public static function getKey(): string;

    public static function getDefault(): ?self;

    public function isDisabled(): bool;
}
```
